message/messaging_service.cc

```diff
@@ -284,6 +284,26 @@ bool messaging_service::is_same_rack(inet_address addr) const {
     return topo.get_rack(addr) == topo.get_rack();
 }
 
+future<> messaging_service::ban_host(locator::host_id id) {
+    return container().invoke_on_all([id] (messaging_service& ms) -> future<> {
```

Comment:
Is invoke_on_all safe with nested coroutines?

Better to make the outer function a coroutine too and use coroutine::lambda to avoid the lambda coroutine fiasco.

Reply:
Unsafety comes from using captures post-yield, which we don't do here.

But the only reason this function is coroutine is because I was yielding in an earlier version of the PR, I no longer do. We can change it to a non-async function at this point. Will follow-up.